### PR TITLE
Wrapped php-http/cache-plugin option blacklisted_paths

### PR DESCRIPTION
**Pull request by nobYsDarling**
Added option for a list of regular expressions to match paths not to be cached.

| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | yes
| Deprecations?   | no
| Related tickets | -
| Documentation   | https://github.com/php-http/documentation/commit/f04eb3b549a8b78d6abab9bac7297139622ac8e1
| License         | MIT

#### What's in this PR?

Adding blacklisted_paths option of cache plugin

#### Why?

New option.

#### Example Usage

``` 
httplug: 
    cache:
        config:
            blacklisted_paths: ['\/path\/not-to-be\/cached']
```

#### Checklist

- [X] Updated CHANGELOG.md to describe BC breaks / deprecations | new feature | bugfix
- [-] Documentation pull request created (if not simply a bugfix) - Already done: https://github.com/php-http/documentation/commit/f04eb3b549a8b78d6abab9bac7297139622ac8e1

### CODE REVIEW

**Comment by dbu** [comment]
thanks! i will wait with merging until we tag a release of the cache plugin.

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -721,6 +721,21 @@ private function createCachePluginNode()
...
+                ->arrayNode('blacklisted_paths')
+                    ->info('An array of regular expression patterns for paths not to be cached. Defaults to an empty array.')
+                    ->defaultValue([])
+                    ->beforeNormalization()
+                        ->castToArray()
+                    ->end()
+                    ->prototype('scalar')
+                        ->validate()
+                            ->ifTrue(function ($v) {
+                                return false === @preg_match('/'.$v.'/', '');
```
Comment (Nyholm) [inline src/DependencyInjection/Configuration.php]: Do we want to protect with `/`?
Why not use `#` as a delimiter? That is way less likely to be added to the URL.

File: tests/Resources/Fixtures/config/full.php [hunk 2]
```diff
@@ -102,6 +102,7 @@
                 'methods' => ['GET'],
                 'cache_key_generator' => null,
                 'respect_response_cache_directives' => ['X-Foo'],
+                'blacklisted_paths' => ['\/path\/not-to-be\/cached']
```
Comment (Nyholm) [inline tests/Resources/Fixtures/config/full.php]: You dont need to escape these slashes with an updated delimiter.

File: tests/Resources/Fixtures/config/full.xml [hunk 3]
```diff
@@ -57,6 +57,7 @@
                 <config default-ttl="42" cache-lifetime="2592000" hash-algo="sha1" cache-key-generator="null">
                     <respect-response-cache-directive>X-Foo</respect-response-cache-directive>
                     <method>GET</method>
+                    <blacklisted-paths>\/path\/not-to-be\/cached</blacklisted-paths>
```
Comment (Nyholm) [inline tests/Resources/Fixtures/config/full.xml]: Same

File: tests/Resources/Fixtures/config/full.yml [hunk 4]
```diff
@@ -73,6 +73,8 @@ httplug:
                 cache_key_generator: null
                 respect_response_cache_directives:
                   - X-Foo
+                blacklisted_paths:
+                  - \/path\/not-to-be\/cached
```
Comment (Nyholm) [inline tests/Resources/Fixtures/config/full.yml]: Same

File: tests/Unit/DependencyInjection/ConfigurationTest.php [hunk 5]
```diff
@@ -233,6 +234,7 @@ public function testSupportsAllConfigFormats(): void
                         'methods' => ['GET'],
                         'cache_key_generator' => null,
                         'respect_response_cache_directives' => ['X-Foo'],
+                        'blacklisted_paths' => ['\/path\/not-to-be\/cached'],
```
Comment (Nyholm) [inline tests/Unit/DependencyInjection/ConfigurationTest.php]: Same

**Comment by Nyholm** [comment]
Could you also update composer.json in require dev to fetch cache-plugin 1.7? (We will soon release that version)

**Pull request merged**